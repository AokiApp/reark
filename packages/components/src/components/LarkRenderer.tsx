import React, { useContext } from "react";
import { LarkApiContext } from "../contexts/LarkApiContext";
import { BlockStoreProvider } from "../contexts/BlockStoreContext";
import { BlockComponent } from "./BlockComponent";

export const LarkRenderer: React.FC = () => {
  const { blocks } = useContext(LarkApiContext);

  if (!blocks || blocks.length === 0) return null;

  const rootBlock =
    blocks.find((block) => block.block_type === 1 && !block.parent_id) ||
    blocks[0];

  if (!rootBlock) return null;

  return (
    <BlockStoreProvider items={blocks}>
      <BlockComponent blockId={rootBlock.block_id} />
    </BlockStoreProvider>
  );
};
