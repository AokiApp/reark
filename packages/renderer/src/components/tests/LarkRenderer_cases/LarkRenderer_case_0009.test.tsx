import { render } from "@testing-library/react";
import { LarkRenderer } from "../../LarkRenderer";

describe("LarkRenderer 複合ケース: 複数ブロック混在", () => {
  it("text, heading, bullet, table など複数ブロックを同時にレンダリングできる", () => {
    const pageBlock = {
      block_id: "page-mix",
      block_type: 1,
      children: ["text-mix", "heading-mix", "bullet-mix", "table-mix"],
      parent_id: "",
    };
    const textBlock = {
      block_id: "text-mix",
      block_type: 2,
      parent_id: "page-mix",
      text: {
        elements: [
          {
            text_run: {
              content: "複合テスト用テキスト",
              text_element_style: { bold: false },
            },
          },
        ],
        style: { align: "left", folded: false },
      },
      children: [],
    };
    const headingBlock = {
      block_id: "heading-mix",
      block_type: 3,
      parent_id: "page-mix",
      heading1: {
        elements: [
          {
            text_run: {
              content: "見出し複合",
              text_element_style: { bold: true },
            },
          },
        ],
        style: { align: 1, folded: false },
      },
      children: [],
    };
    const bulletBlock = {
      block_id: "bullet-mix",
      block_type: 12,
      parent_id: "page-mix",
      bullet: {
        elements: [
          {
            text_run: {
              content: "リスト複合",
              text_element_style: { bold: false },
            },
          },
        ],
        style: { align: 1, folded: false },
      },
      children: [],
    };
    const tableBlock = {
      block_id: "table-mix",
      block_type: 18,
      parent_id: "page-mix",
      table: {
        cells: [
          [{ text: "セルA" }, { text: "セルB" }],
          [{ text: "セルC" }, { text: "セルD" }],
        ],
        property: {},
      },
      children: [],
    };

    const blocks = [
      pageBlock,
      textBlock,
      headingBlock,
      bulletBlock,
      tableBlock,
    ];

    const { container } = render(<LarkRenderer initialData={{ blocks }} />);
    expect(container).toMatchSnapshot();
  });
});
