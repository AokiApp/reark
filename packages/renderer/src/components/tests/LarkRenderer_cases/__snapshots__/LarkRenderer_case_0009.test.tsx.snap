// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LarkRenderer 複合ケース: 複数ブロック混在 text, heading, bullet, table など複数ブロックを同時にレンダリングできる 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-page"
    >
      <div
        class="reark-page-title"
      >
        <div
          class="reark-text-block"
          style="text-align: left;"
        />
      </div>
      <div
        class="reark-block"
      >
        <div
          class="reark-text-block"
          style="text-align: left;"
        >
          <span
            class="reark-text"
          >
            複合テスト用テキスト
          </span>
        </div>
      </div>
      <div
        class="reark-block"
      >
        <div
          class="reark-heading reark-heading--1"
        >
          <div
            class="reark-text-block"
            style="text-align: left;"
          >
            <span
              class="reark-text--bold reark-text"
            >
              見出し複合
            </span>
          </div>
        </div>
      </div>
      <div
        class="reark-block"
      >
        <div
          class="reark-unordered-list"
        >
          <div
            class="reark-text-block"
            style="text-align: left;"
          >
            <span
              class="reark-text"
            >
              リスト複合
            </span>
          </div>
        </div>
      </div>
      <div
        class="reark-unsupported-block"
      >
        <strong>
          Unsupported block type: 
          18
        </strong>
      </div>
    </div>
  </div>
</div>
`;
